scala-package/native/src/main/native/ml_dmlc_mxnet_native_c_api.cc

```diff
@@ -2301,7 +2301,9 @@ JNIEXPORT jint JNICALL Java_ml_dmlc_mxnet_LibInfo_mxCustomOpRegister
                   ptrsArr, (jsize)0, (jsize)size, reinterpret_cast<jlong*>(ptrs));
 #if MXNET_USE_CUDA
                 mxnet::NDArray* tmp = reinterpret_cast<mxnet::NDArray*>(ptrs[0]);
-                CUDA_CALL(cudaSetDevice(tmp->ctx().dev_id));
+                if (tmp->ctx().dev_type != mxnet::Context::kCPU) {
```

Comment:
== kGPU

Reply:
Revise the conditional statement to be more readable. Thanks for the comment.